### ISSUE TITLE
Task 4.3: Implement SAE Training

### ISSUE THREAD

**Issue by juancadile** [juancadile/empathy-probes#13]
## Description
Train Sparse Autoencoders (SAEs) to identify monosemantic empathy features across model families.

## Models for SAE Training (start with 2-3)

| Model | Target Layers | Hidden Dim | SAE Expansion |
|-------|---------------|------------|---------------|
| Qwen-2.5-32B | top-3 AUROC | 5120 | 4x, 8x |
| Yi-1.5-34B | top-3 AUROC | 7168 | 4x, 8x |
| Mistral-Small-24B | top-3 AUROC | 5120 | 4x, 8x |

## SAE Architecture
```python
class TopKSAE(nn.Module):
    def __init__(self, d_model, d_sae, k=32):
        self.encoder = nn.Linear(d_model, d_sae)
        self.decoder = nn.Linear(d_sae, d_model)
        self.k = k  # Top-k sparsity
    
    def forward(self, x):
        # Encode
        z = self.encoder(x)
        # Top-k activation
        topk_vals, topk_idx = torch.topk(z, self.k)
        z_sparse = torch.zeros_like(z).scatter(-1, topk_idx, topk_vals)
        # Decode
        return self.decoder(z_sparse), z_sparse
```

## Tasks
- [ ] Write `train_sae.py` using SAELens or custom TopK implementation
- [ ] Train on selected layers (top AUROC layers)
- [ ] SAE dimensions: 4x, 8x expansion (e.g., 5120 → 20480, 40960)
- [ ] Top-k sparsity: k ∈ {16, 32, 64}
- [ ] Identify monosemantic empathy features
- [ ] Feature ablation: remove SAE features → measure probe change
- [ ] Visualize top empathy-correlated features

## New file
`src/train_sae.py`

## Dependencies
- SAELens (recommended) or custom implementation
- Requires stored token-level activations (~50GB per model)

## Compute Notes
- **MUST use BF16 precision** (not FP8) for accurate gradients
- 20-40 hours for 2-4 SAEs per model
- Checkpoint frequently

## Expected Output
Figure 10: SAE feature analysis
- Feature activation histograms
- Top features correlated with empathy
- Feature ablation effects

## Priority
Optional (Priority 3) but HIGH IMPACT if implemented - publishable standalone result
